mlir-sycl/test/Conversion/SYCLToLLVM/sycl-methods-to-llvm.mlir

```diff
@@ -1,229 +1,219 @@
-// CHECK-NEXT:      %[[VAL_11:.*]] = llvm.load %[[VAL_10]] : !llvm.ptr<ptr<i32, 1>>
-// CHECK-NEXT:      %[[VAL_12:.*]] = llvm.getelementptr inbounds %[[VAL_11]][%[[VAL_9]]] : (!llvm.ptr<i32, 1>, i64) -> !llvm.ptr<i32, 1>
-// CHECK-NEXT:      llvm.return %[[VAL_12]] : !llvm.ptr<i32, 1>
-// CHECK-NEXT:    }
+// CHECK-NEXT:      %[[VAL_9:.*]] = llvm.sub %[[VAL_1]], %[[VAL_8]]  : i64
+// CHECK-NEXT:      %[[VAL_10:.*]] = llvm.getelementptr inbounds %[[VAL_0]][0, 1, 0] : (!llvm.ptr) -> !llvm.ptr, !llvm.struct<"class.sycl::_V1::accessor.1", {{.*}}>
+// CHECK-NEXT:      %[[VAL_11:.*]] = llvm.load %[[VAL_10]] : !llvm.ptr -> !llvm.ptr<1>
+// CHECK-NEXT:      %[[VAL_12:.*]] = llvm.getelementptr inbounds %[[VAL_11]]{{\[}}%[[VAL_9]]] : (!llvm.ptr<1>, i64) -> !llvm.ptr<1>, i32
+// CHECK-NEXT:      llvm.return %[[VAL_12]] : !llvm.ptr<1>
 func.func @test(%acc: memref<?x!sycl_accessor_1_i32_rw_gb>) -> memref<?xi32, 1> {
   %0 = sycl.accessor.get_pointer(%acc) { ArgumentTypes = [memref<?x!sycl_accessor_1_i32_rw_gb>], FunctionName = @"get_pointer", MangledFunctionName = @"get_pointer", TypeName = @"accessor" }  : (memref<?x!sycl_accessor_1_i32_rw_gb>) -> memref<?xi32, 1>
   return %0 : memref<?xi32, 1>
 }
 
 // -----
 
-//===-------------------------------------------------------------------------------------------------===//
+//===----------------------------------------------------------------------===//
 // sycl.accessor.get_range
-//===-------------------------------------------------------------------------------------------------===//
+//===----------------------------------------------------------------------===//
 
 !sycl_id_1_ = !sycl.id<[1], (!sycl.array<[1], (memref<1xi64>)>)>
 !sycl_range_1_ = !sycl.range<[1], (!sycl.array<[1], (memref<1xi64>)>)>
 !sycl_accessor_impl_device_1_ = !sycl.accessor_impl_device<[1], (!sycl_id_1_, !sycl_range_1_, !sycl_range_1_)>
 !sycl_accessor_1_i32_rw_gb = !sycl.accessor<[1, i32, read_write, global_buffer], (!sycl_accessor_impl_device_1_, !llvm.struct<(ptr<i32, 1>)>)>
 
 // CHECK-LABEL:   llvm.func @test(
-// CHECK-SAME:                    %[[VAL_0:.*]]: !llvm.ptr<[[ACCESSOR1:.*]]>) -> !llvm.[[RANGE1:.*]] {
-// CHECK-NEXT:      %[[VAL_1:.*]] = llvm.getelementptr inbounds %[[VAL_0]][0, 0, 1] : (!llvm.ptr<[[ACCESSOR1]]>) -> !llvm.ptr<[[RANGE1]]>
-// CHECK-NEXT:      %[[VAL_2:.*]] = llvm.load %[[VAL_1]] : !llvm.ptr<[[RANGE1]]>
-// CHECK-NEXT:      llvm.return %[[VAL_2]] : !llvm.[[RANGE1]]
-// CHECK-NEXT:    }
+// CHECK-SAME:                    %[[VAL_0:.*]]: !llvm.ptr) -> !llvm.struct<"class.sycl::_V1::range.1", {{.*}}> {
+// CHECK-NEXT:      %[[VAL_1:.*]] = llvm.getelementptr inbounds %[[VAL_0]][0, 0, 1] : (!llvm.ptr) -> !llvm.ptr, !llvm.struct<"class.sycl::_V1::accessor.1", {{.*}}>
+// CHECK-NEXT:      %[[VAL_2:.*]] = llvm.load %[[VAL_1]] : !llvm.ptr -> !llvm.struct<"class.sycl::_V1::range.1", {{.*}}>
+// CHECK-NEXT:      llvm.return %[[VAL_2]] : !llvm.struct<"class.sycl::_V1::range.1", {{.*}}>
```

Comment:
```suggestion
// CHECK-SAME:                    %[[VAL_0:.*]]: !llvm.ptr) -> [[RANGE1:.*]] {
// CHECK-NEXT:      %[[VAL_1:.*]] = llvm.getelementptr inbounds %[[VAL_0]][0, 0, 1] : (!llvm.ptr) -> !llvm.ptr, !llvm.struct<"class.sycl::_V1::accessor.1", {{.*}}>
// CHECK-NEXT:      %[[VAL_2:.*]] = llvm.load %[[VAL_1]] : !llvm.ptr -> [[RANGE1]]
// CHECK-NEXT:      llvm.return %[[VAL_2]] : [[RANGE1]]
```

Reply:
If you agree that is more readable, then similar changes for the other types. No strong preference.

Reply:
I had considered that, but decided against for mainly three reasons:
* The current version actually checks the type, whereas the substitution block would also allow other, wrong types to be matched. 
* With opaque pointers, the type matched by the substitution block in the old test isn't part of the function signature anymore, so the definition of the substitution would happen later than it did before.
* Doing this replacement requires quite a bit of work. From the experience of updating all these tests as part of this PR, I prefer the tests to be close to the output of `generate-test-checks.py`, to allow quick updating of tests in case of changes. The new version of the checks can be generated from `generate-test-checks.py` with just two replace-all (`CHECK:` -> `CHECK-NEXT`, `!llvm.struct<"class.sycl::_V1::range.1", very_long_detail_type>` -> `!llvm.struct<"class.sycl::_V1::range.1", {{.*}}>`) and a handful of `DAG` insertions. 

If you agree, I would leave as is.

Reply:
Yes, I agree.